pig/src/main/java/org/jboss/pnc/bacon/pig/impl/utils/FileCache.java

```diff
@@ -0,0 +1,71 @@
+package org.jboss.pnc.bacon.pig.impl.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.ConcurrentMap;
+
+import org.mapdb.DB;
+import org.mapdb.DBMaker;
+import org.mapdb.Serializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileCache {
+
+    private static final Logger log = LoggerFactory.getLogger(FileCache.class);
+    private static final int MAX_SIZE_GB = 5;
+
+    private DB db;
+    private ConcurrentMap<String, byte[]> cachedMap;
+
+    public FileCache(Path cacheFile) {
+
+        // if cache file parent folder doesn't exist, create it
+        if (!Files.exists(cacheFile.getParent())) {
+            try {
+                Files.createDirectories(cacheFile.getParent());
+            } catch (IOException e) {
+                log.error("Unable to create directory {}", cacheFile.getParent(), e);
```

Comment:
Hey @thescouser89 If the parent folder doesn't exist and the creation failed, shouldn't then return at this point?

Reply:
Oh yeah you're right! Completely missed this. It should rethrow the error.